src/Raven.Server/Documents/ChangesClientConnection.cs

```diff
@@ -574,7 +583,64 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ms))
                             break;
 
                         ms.TryGetBuffer(out ArraySegment<byte> bytes);
-                        await _webSocket.SendAsync(bytes, WebSocketMessageType.Text, true, _disposeToken);
+
+                        var sendTask = _webSocket.SendAsync(bytes, WebSocketMessageType.Text, true, _disposeToken);
+                        var sendSp = Stopwatch.StartNew();
+
+                        while (true)
+                        {
+                            var waitTask = TimeoutManager.WaitFor(TimeSpan.FromSeconds(5), _disposeToken);
+
+                            var waitResult = Task.WaitAny(sendTask, waitTask);
+                            if (waitResult == 0)
+                            {
+                                await sendTask;
+                                break;
+                            }
+
+                            var isLowMemory = _lowMemoryFlag.IsRaised();
+                            switch (isLowMemory)
+                            {
+                                case true:
+                                    if (_sendQueue.Count < 16 * 1024)
+                                    {
+                                        // we are in low memory state and the number of messages in the queue is reasonable.
```

Comment:
don't we need to add `await sendTask;` here?

Reply:
I start waiting here for another 5 seconds.
https://github.com/ravendb/ravendb/pull/15763/files#diff-b44215a140366567f4495107fe9a77190b5f40ed97cc1ead42a7d44f3311d961R591

Reply:
yes, but what happens to current `sendTask`? You are creating another one, which means that you can send via WebSocket in parallel, which is not supported?

Reply:
The task is created before the `while`

Reply:
oh, yes. Somehow I focused on the outer while.